ocaml/parsing/extensions_parsing.ml

```diff
@@ -136,14 +136,17 @@ module type AST_parameters = sig
       [fun tm -> tm.pCAT_loc] for the appropriate syntactic category [CAT]. *)
   val location : ast -> Location.t
 
-  (** Turn an [ast_desc] into an [ast] by adding the appropriate metadata *)
+  (** Turn an [ast_desc] into an [ast] by adding the appropriate metadata.  When
+      creating [ast] nodes afresh for an extension, the location should be
+      omitted; in this case, it will default to [!Ast_helper.default_loc], which
+      should be [ghost]. *)
   val wrap_desc :
-    loc:Location.t -> attrs:Parsetree.attributes -> ast_desc -> ast
+    ?loc:Location.t -> attrs:Parsetree.attributes -> ast_desc -> ast
 
   (** How to construct an extension node for this AST (something of the shape
       [[%name]] or [[%%name]], depending on the AST).  Should just be
-      [Ast_helper.CAT.extension] for the appropriate syntactic category
-      [CAT]. *)
+      [Ast_helper.CAT.extension] for the appropriate syntactic category [CAT].
+      (This means that [?loc] should default to [!Ast_helper.default_loc.].) *)
```

Comment:
By this logic, maybe `make_extension_node` should just not carry the implicit parameter?

Reply:
Because that's just in the `.ml` file, I prioritized making it *really* just the same as the function from `Ast_helper`